### PR TITLE
[v6r16] Monitoring system

### PR DESCRIPTION
**Pull request by zmathe**

### CODE REVIEW

**Comment by fstagni** [comment]
This PR targets integration

File: Core/Base/ElasticDB.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  """
+  
+  ########################################################################
+  def __init__( self, dbname, fullName ):
+    """ c'tor
+
+    :param self: self reference
+    :param str dbName: name of the database for example: MonitoringDB
+    :param str fullName: The full name of the database for example: 'Monitoring/MonitoringDB'
+    :param bool debug: save the debug information to a file
```
Comment (fstagni) [inline Core/Base/ElasticDB.py]: There's no `debug`

File: Core/Base/ElasticDB.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+      raise RuntimeError( 'Can not connect to DB %s, exiting...' % self.dbName )
+
+
+    self.log.info( "==================================================" )
+    self.log.info( "Host:           " + self.__dbHost )
+    self.log.info( "Port:           " + str( self.__dbPort ) )
+    self.log.info( "ClusterName:    " + self.clusterName )
+    self.log.info( "==================================================" )
+  
+  ########################################################################
```
Comment (fstagni) [inline Core/Base/ElasticDB.py]: I remain of my idea that there's no need for any setter/getter methods

File: Core/Utilities/ElasticSearchDB.py [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+  
+  def deleteIndex( self, indexName ):
+    """
+    :param str indexName the name of the index to be deleted...
+    """
+    try:
+      retVal = self.__client.indices.delete( indexName )
+    except  NotFoundError  as e:
+      return S_ERROR( DErrno.EELNOFOUND, e )
+    except ValueError as e:
```
Comment (fstagni) [inline Core/Utilities/ElasticSearchDB.py]: When would this be triggered?

File: Core/Utilities/Plotting/ObjectLoader.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
+"""
+It is used to load classes from a specific system.
```
Comment (fstagni) [inline Core/Utilities/Plotting/ObjectLoader.py]: What was wrong with the `DIRAC.Core.Utilities.ObjectLoader` module?

File: Core/Utilities/Plotting/Plots.py [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+  Tis generate an image with a specific error message.
+  :param str fileName name of the file
+  :param list data data
+  :param dict metadata metadata information
+  """
+  try:
+    with open( fileName, "wb" ) as fn:
+      text = "No data for this selection for the plot: %s" % metadata['title']
+      textGraph( text, fn, metadata )
+      fn.close()
```
Comment (fstagni) [inline Core/Utilities/Plotting/Plots.py]: using `with`, no need to explicitly call `fn.close()`

File: Core/Utilities/Plotting/Plots.py [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+  :param dict metadata extra information used to create the plot.
+  """
+  try:
+    with open( fileName, "wb" ) as fn:
+      checkMetadata( metadata )
+      for key, value in ( ( 'sort_labels', 'sum' ), ( 'legend_unit', '%' ) ):
+        if key not in metadata:
+          metadata[ key ] = value
+      barGraph( data, fn, **metadata )
+      fn.close()
```
Comment (fstagni) [inline Core/Utilities/Plotting/Plots.py]: same... no need

File: Core/Utilities/Plotting/Plots.py [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+  try:
+    with open( fileName, "wb" ) as fn:
+      checkMetadata( metadata )
+      metadata[ 'legend' ] = False
+      #HACK: Pad a bit to the left until the proper padding is calculated
+      maxKeyLength = max( [ len( key ) for key in data ] )
+      metadata[ 'sort_labels' ] = 'alpha'
+      metadata[ 'plot_left_padding' ] = int( maxKeyLength * 2.5 )
+      qualityGraph( data, fn, **metadata )
+      fn.close()
```
Comment (fstagni) [inline Core/Utilities/Plotting/Plots.py]: same... and in other cases in this module too

File: Core/Utilities/ElasticSearchDB.py (hunk 3 above)
Comment (zmathe) [inline Core/Utilities/ElasticSearchDB.py]: If the index does not exists and try to delete

File: Core/Utilities/Plotting/ObjectLoader.py (hunk 4 above)
Comment (zmathe) [inline Core/Utilities/Plotting/ObjectLoader.py]: this is specific for Accounting and Monitoring

**Pull request merged**